scipy/stats/_stats_py.py

```diff
@@ -1817,20 +1817,97 @@ def jarque_bera(x, *, axis=None):
            homoscedasticity and serial independence of regression residuals",
            6 Econometric Letters 255-259.
+    .. [2] Shapiro, S. S., & Wilk, M. B. (1965). An analysis of variance test
+           for normality (complete samples). Biometrika, 52(3/4), 591-611.
+    .. [3] B. Phipson and G. K. Smyth. "Permutation P-values Should Never Be
+           Zero: Calculating Exact P-values When Permutations Are Randomly
+           Drawn." Statistical Applications in Genetics and Molecular Biology
+           9.1 (2010).
 
     Examples
     --------
+    Suppose we wish to infer from measurements whether the weights of adult
+    human males in a medical study are not normally distributed [2]_.
+    The weights (lbs) are recorded in the array ``x`` below.
+
     >>> import numpy as np
+    >>> x = np.array([148, 154, 158, 160, 161, 162, 166, 170, 182, 195, 236])
+
+    The Jarque-Bera test begins by computing a statistic based on the sample
+    skewness and kurtosis.
+
     >>> from scipy import stats
-    >>> rng = np.random.default_rng()
-    >>> x = rng.normal(0, 1, 100000)
-    >>> jarque_bera_test = stats.jarque_bera(x)
-    >>> jarque_bera_test
-    Jarque_beraResult(statistic=3.3415184718131554, pvalue=0.18810419594996775)
-    >>> jarque_bera_test.statistic
-    3.3415184718131554
-    >>> jarque_bera_test.pvalue
-    0.18810419594996775
+    >>> res = stats.jarque_bera(x)
+    >>> res.statistic
+    6.982848237344646
+    >>> s = stats.skew(x)  # calculate the sample skewness
+    >>> k = stats.kurtosis(x)  # calculate the sample kurtosis
+    >>> len(x)/6 * (s**2 + k**2/4)  # calculate the Jarque-Bera statistic
+    6.982848237344646
+
+    Because the normal distribution has zero skewness and zero
+    ("excess" or "Fisher") kurtosis, the value of this statistic tends to be
+    low for samples drawn from a normal distribution.
+
+    The test is performed by comparing the observed value of the statistic
+    against the null distribution: the distribution of statistic values derived
+    under the null hypothesis that the sample *is* normally distributed.
+    For the Jarque-Bera test, the null distribution for very large samples is
+    the chi-squared distribution with two degrees of freedom.
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots()
+    >>> dist = stats.chi2(df=2)
+    >>> jb_val = np.linspace(0, 10, 100)
+    >>> ax.plot(jb_val, dist.pdf(jb_val))
+    >>> ax.set_title("PDF of Jarque-Bera Null Distribution")
+    >>> ax.set_xlabel("statistic")
+    >>> ax.set_ylabel("probability density")
+    >>> plt.show()
+
+    The comparison is quantified by the p-value: the proportion of values in
+    the null distribution greater than or equal to the observed value of the
+    statistic.
+
+    >>> dist.sf(res.statistic)
+    0.03045746622458189
+    >>> res.pvalue
+    0.03045746622458189
+
+    If the p-value is small - that is, if there is a low probability of
```

Comment:
Small being relative, I think we need to say with respect to which level (5%?)

Reply:
That is always up to the researcher. I always avoid suggesting a level. But I will make it explicit that it is the researcher's choice and perhaps note the convention of a few fields.

Reply:
Totally, and in particule physics 3% might still be high. That's what I had in mind with saying that we need a reference to say "small"